packages/lib-react-components/src/AnimatedNumber/AnimatedNumber.js

```diff
@@ -4,80 +4,94 @@ import { interpolate } from '@visx/vendor/d3-interpolate'
 import { select } from 'd3'
 import { useEffect, useRef, useState } from 'react'
 
-const initialValue = 0
-
 let prefersReducedMotion
 const isBrowser = typeof window !== 'undefined'
 if (isBrowser) {
-  prefersReducedMotion = window.matchMedia('(prefers-reduced-motion: reduce)').matches
+  prefersReducedMotion = () => window.matchMedia('(prefers-reduced-motion: reduce)').matches
 }
+export { prefersReducedMotion }
 
-function AnimatedNumber({ duration = 1000, value }) {
-  const numRef = useRef(null)
-  const [animated, setAnimated] = useState(false)
+function formatValue(num) {
+  return format(',d')(num)
+}
 
-  function animateValue() {
-    select(numRef.current)
-      .data([value])
-      .transition()
-      .duration(duration)
-      .textTween(() => {
-        const interpolator = interpolate(initialValue, value)
-        return t => {
-          const interpolatedValue = interpolator(t)
-          if (interpolatedValue === value) setAnimated(true) // animation complete!
-          const niceValue = formatValue(interpolatedValue)
-          return niceValue
-        }
-      })
+/**
+ * Use d3 to animate `element` from `initialValue` to `value` over `duration` in milliseconds.
+ * @param {HTMLElement} element 
+ * @param {number} initialValue 
+ * @param {number} value
+ * @param {number} duration 
+ * @returns a Promise that resolves once the animation is complete.
+ */
+function animateValue(element, initialValue, value, duration) {
+  if (value === initialValue) {
+    return
   }
+  const interpolator = interpolate(initialValue, value)
+  return select(element)
+    .data([value])
+    .transition()
+    .duration(duration)
+    .textTween(() => t => formatValue(interpolator(t)))
+    .end()
```

Comment:
`transition().end()` returns a Promise that resolves when `transition()` completes, and rejects if the animation is interrupted. That's super useful.
https://observablehq.com/@d3/transition-end